{
 "cells": [
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----\n",
    "# Study aims\n",
    "----\n",
    "> Here, we [...] examine the medusa (free-swimming jellyfish) stage of the hydrozoan Clytia\n",
    "> hemisphaerica, with dual motivations. First, Clytia is a powerful, emerging model system spanning\n",
    "> multiple fields, from evolutionary and developmental biology to regeneration and neuroscience.\n",
    "> While previous work has characterized a number of cell types in the Clytia medusa, a\n",
    "> whole-organism atlas of transcriptomic cell types has been lacking. Such an atlas is a critical\n",
    "> resource for the Clytia community and an important addition to the study of cell types across\n",
    "> animal phylogeny.\n",
    "> \n",
    "> Second, emerging multiplexing techniques present new opportunities for system-level studies of\n",
    "> cell types and their changing states at unprecedented resolution in whole organisms. The Clytia\n",
    "> medusa offers an appealing platform for pioneering these studies. It is small, transparent, and\n",
    "> has simple tissues and organs, stem cell populations actively replenishing many cell types in\n",
    "> mature animals, and remarkable regenerative capacity. Furthermore, the 1-cm-diameter adult medusae\n",
    "> used in this study contain on the order of $10^5$ cells, making it possible to sample cells\n",
    "> comprehensively across a whole animal in a cost-effective manner using current scRNA-seq\n",
    "> technology. In this study, we generate a cell atlas for the Clytia medusa while simultaneously\n",
    "> performing a whole-organism perturbation study, providing the first medusa single-cell dataset and\n",
    "> an examination of changing cell states across the organism. Our work also provides a\n",
    "> proof-of-principle for perturbation studies in nontraditional model organisms, using multiplexing\n",
    "> technology and a reproducible workflow with lessened reliance on functional annotation, from the\n",
    "> experimental implementation to the data processing and analysis.\n",
    "\n",
    "# Study design\n",
    "\n",
    "> Experimental design of the starvation experiment showing (1) images of control versus 4-day\n",
    "> starved animals (scale bars, 0.5 cm), (2) dissociation of individual medusa and chemical tagging\n",
    "> of cells with ClickTags to enable multiplexed scRNA-seq, (3) pooling of cells and library\n",
    "> generation from lysed cells to generate (4) sequencing libraries for the multiplexed cDNA and\n",
    "> ClickTag data and create (5) single-cell resolved gene expression count matrices from all animals:\n",
    "\n",
    "<div>\n",
    "<img src=\"Fig1.JPG\" width=\"600\"/>\n",
    "</div>"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "<!-- # Task 1 - experimental design\n",
    "\n",
    "#### 1. Why is single-cell RNA-seq the method of choice?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 2. Why is _Clytia_ a good model system?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 3. What are the aims of the study?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 4. How will the experimental design achieve the stated aims of the study?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 5. Identify the biological and technical replicates.\n",
    "\n",
    "<span style=\"color:green\">Answer</span> -->\n"
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----\n",
    "# Task 1 - basic single-cell RNA-seq analysis\n",
    "----\n",
    "\n",
    "## Theory:\n",
    "\n",
    "#### 1. What are the basic steps for the analysis of single-experiment scRNA-seq data?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 2. What do we want to achieve with normalisation? Name as many methods for normalisation as you can.\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 3. Explain a normalisation method of your choice using your own words.\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 4. What do we want to achieve with variance stabilisation? What are two methods to achieve it?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 5. Why do we perform dimensionality reduction? Why might PCA be a good choice?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 6. Why do we cluster? Does clustering make sense in the context of scRNA-seq data?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 7. What is a way to connect clustering to cell types?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "## Exercise:\n",
    "\n",
    "Read in the data from the fed animals of batch 1. Preprocess and integrate the data and cluster to\n",
    "reveal putative cell types. Try to explain your steps and decisions so that an outside observer can\n",
    "follow the analysis."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----\n",
    "# Task 2 - Integration\n",
    "----\n",
    "\n",
    "## Theory:\n",
    "\n",
    "#### 1. What are possible sources of batch effect in scRNA-seq data?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 2. Why is batch effect a problem in scRNA-seq data?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 3. Name some strategies to remove batch effect.\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "## Exercise:\n",
    "\n",
    "Read in the data from the fed animals of batch 0. Integrate with the cells from batch 1. Demonstrate\n",
    "that the integration was successful. Try to explain your steps and decisions so that an outside\n",
    "observer can follow the analysis."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----\n",
    "# Task 3 - Advanced scRNA-seq analysis\n",
    "----\n",
    "\n",
    "## Theory:\n",
    "\n",
    "#### 1. What is the evolutionary definition of cell types?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 2. What predictions does the theory make about the data?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "\n",
    "## Exercise:\n",
    "\n",
    "Identify the cell cluster assigned to you by using its distinct genome partition.\n",
    "\n",
    "The evolutionary definition of cell types implies hierarchical relationships between cell types\n",
    "within and between species. Cluster the integrated data and make these hierarchical relationships\n",
    "visible.\n",
    "\n",
    "Try to explain your steps and decisions so that an outside observer can follow the analysis."
   ]
  },
  {
   "attachments": {},
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "----\n",
    "# Task 4 - Cross-species comparisons\n",
    "----\n",
    "\n",
    "## Theory:\n",
    "\n",
    "#### 1. What are possible problems that make cross-species comparisons challenging?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "#### 2. What predictions does the evolutionary cell type theory make about cross-species comparisons?\n",
    "\n",
    "<span style=\"color:green\">Answer</span>\n",
    "\n",
    "\n",
    "## Exercise:\n",
    "\n",
    "Get the \n",
    "\n",
    "Try to explain your steps and decisions so that an outside observer can follow the analysis."
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "ascc23",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.16"
  },
  "orig_nbformat": 4
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
